routers/user/auth.go

```diff
@@ -664,8 +664,25 @@ func LinkAccount(ctx *context.Context) {
 		return
 	}
 
-	ctx.Data["user_name"] = gothUser.(goth.User).NickName
-	ctx.Data["email"] = gothUser.(goth.User).Email
+	uname := gothUser.(goth.User).NickName
+	email := gothUser.(goth.User).Email
+	ctx.Data["user_name"] = uname
+	ctx.Data["email"] = email
+
+	if len(email) != 0 {
+		// ignoring the "user not found" error
+		u, _ := models.GetUserByEmail(email)
+		if u != nil {
+			ctx.Data["user_exists"] = true
+		}
+	} else if len(uname) != 0 {
+		// ignoring the "user not found" error
+		u, _ := models.GetUserByName(uname)
+		fmt.Println("Error retrieving username:", err)
```

Comment:
Won't compile with that line

Reply:
Oops, I had the correct files on disk, but missed the last `git add`. Will fix.